### ISSUE TITLE
false negative on tuple with incorrect length when instantiated with `tuple`

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#15171]
```py
foo = tuple[int, int]((1, 2, 3)) # no error
boo: tuple[int, int] = (1, 2, 3) # error: Incompatible types in assignment (expression has type "Tuple[int, ...]", variable has type "Tuple[int, int]")  [assignment]
```
[playground](https://mypy-play.net/?mypy=latest&python=3.11&flags=show-error-context%2Cshow-column-numbers%2Cshow-error-codes%2Cstrict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cno-implicit-optional%2Cno-implicit-reexport%2Cno-strict-optional%2Cstrict-equality%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=306a07cfad71f4042e0225c535417502)

**Comment by erictraut** [comment]
This is because of the way the `tuple` constructor is defined in typeshed. A static type checker would need to override the typeshed definition and provide special-case logic to catch the above issue. That's perhaps a reasonable feature request, but I don't think the current behavior is a bug.

**Comment by urnest** [comment]
current master (10f18a82b612b6127659cd64aa60c10b9cc7a904) gives:
/home/xju/tmp/xxx.py:1:23: error: Argument 1 to "tuple" has incompatible type "tuple[int, int, int]"; expected "tuple[int, int]"  [arg-type]
/home/xju/tmp/xxx.py:2:24: error: Incompatible types in assignment (expression has type "tuple[int, int, int]", variable has type "tuple[int, int]")  [assignment]
Found 2 errors in 1 file (checked 1 source file)

... is that what we need in this case?

**Comment by brianschubert** [comment]
Looks like this was fixed in 6c24ea66e20166964aa5d42e28fda5b4b69f44b1 (#17235), released with v1.11.0
